riscv-pages/src/page.rs

```diff
@@ -88,6 +88,54 @@ impl PageSize {
     pub fn num_pages_contained_in(&self, val: u64) -> u64 {
         val >> self.page_shift()
     }
+
+    /// Return the maximum-size page aligned at an address
+    pub fn max_page_alignment(val: u64) -> Option<PageSize> {
+        match val.trailing_zeros() {
+            0..=11 => None,
+            12..=20 => Some(PageSize::Size4k),
+            21..=29 => Some(PageSize::Size2M),
+            30..=38 => Some(PageSize::Size1G),
+            _ => Some(PageSize::Size512G),
+        }
+    }
+
+    /// Iterate down page size
+    pub fn down_size(&self) -> Option<PageSize> {
+        match self {
+            PageSize::Size512G => Some(PageSize::Size1G),
+            PageSize::Size1G => Some(PageSize::Size2M),
+            PageSize::Size2M => Some(PageSize::Size4k),
+            PageSize::Size4k => None,
+        }
+    }
+
+    /// Iterate up one page size
+    pub fn up_size(&self) -> PageSize {
```

Comment:
This should probably return an `Option<>` as well, with `up_size(Size512G)` returning `None`.